test/close.test.js

```diff
@@ -495,3 +495,50 @@ test('shutsdown while keep-alive connections are active (non-async, custom)', t
     })
   })
 })
+
+test('preClose callback', t => {
```

Comment:
Can we test with multiple `preClose` hook and mixed `callback` with `async-await` to ensure the execution order.

```js
test('preClose execution order', t => {
  t.plan(5)
  async function sleep(ms) {
    return new Promise((resolve) => {
      setTimeout(resolve, ms)
    })
  }
  const fastify = Fastify()
  const order = []
  fastify.addHook('onClose', onClose)
  function onClose (instance, done) {
    t.type(fastify, instance)
    t.same(order, [1, 2, 3])
    done()
  }

  fastify.addHook('preClose', (done) => {
    setTimeout(function () {
      order.push(1)
      done()
    }, 200)
  })

  fastify.addHook('preClose', async () => {
    await sleep(100)
    order.push(2)
  })

  fastify.addHook('preClose', (done) => {
    setTimeout(function () {
      order.push(3)
      done()
    }, 100)
  })

  fastify.listen({ port: 0 }, err => {
    t.error(err)

    fastify.close((err) => {
      t.error(err)
      t.ok('close callback')
    })
  })
})
```

Reply:
added